pkg/gatewayserver/io/io.go

```diff
@@ -265,7 +269,22 @@ func (c *Connection) ScheduleDown(path *ttnpb.DownlinkPath, msg *ttnpb.DownlinkM
 	if err != nil {
 		return 0, err
 	}
-	phy, err := band.GetByID(c.fp.BandID)
+	var fp *frequencyplans.FrequencyPlan
+	for _, gtwFP := range c.fps {
+		reqFP, err := c.fpStore.GetByID(request.FrequencyPlanID)
+		if err != nil {
+			return 0, err
+		}
+		if gtwFP.BandID == reqFP.BandID {
+			fp = gtwFP
```

Comment:
Also what does this mean; the frequency plan matches if the band matches?

Reply:
No indeed. The frequency plan ID should match. Fixed.